File: Examples/VisualBasic/ModifyingAndConvertingImages/JPEG/ReadAndModifyJpegEXIFTags.vb
Imports System.Reflection
Imports Aspose.Imaging.Exif
Imports Aspose.Imaging.FileFormats.Jpeg

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.JPEG
    Class ReadAndModifyJpegEXIFTags
        Public Shared Sub Run()
            ' ExStart:ReadAndModifyJpegEXIFTags
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_JPEG()

            ' Load an image using the factory method Load exposed by Image class
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("aspose-logo.jpg"))
                ' Initialize an object of ExifData and fill it will image's EXIF information and Check if image has any EXIF entries defined
                Dim exif As ExifData = DirectCast(image__1, JpegImage).ExifData
                ' Check if image has any EXIF entries defined
                If exif IsNot Nothing Then
                    ' In order to get all EXIF tags, first get the Type of EXIF object, Get all properties of EXIF object into an array and Iterate over the EXIF properties
                    Dim type As Type = exif.[GetType]()
                    Dim properties As PropertyInfo() = type.GetProperties()
                    For Each [property] As PropertyInfo In properties
                        ' Display property name and its value
                        Console.WriteLine([property].Name + ":")
                        Console.WriteLine([property].GetValue(exif, Nothing))
                    Next
                End If
            End Using
            ' ExEnd:ReadAndModifyJpegEXIFTags
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/JPEG/ReadAndModifyJpegEXIFTags.cs
using System;
using System.Reflection;
using Aspose.Imaging.Exif;
using Aspose.Imaging.FileFormats.Jpeg;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download the Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.JPEG
{
    class ReadAndModifyJpegEXIFTags
    {
        public static void Run()
        {
            //ExStart:ReadAndModifyJpegEXIFTags
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_JPEG();

            Console.WriteLine("Running example ReadAndModifyJpegEXIFTags");
            // Load an image using the factory method Load exposed by the Image class.
            using (Image image = Image.Load(dataDir + "aspose-logo.jpg"))
            {
                // Initialize an ExifData object and fill it with the image's EXIF information. Then check if the image has any EXIF entries defined.
                ExifData exif = ((JpegImage)image).ExifData;
                if (exif != null)
                {
                    // To get all EXIF tags, first obtain the type of the EXIF object,
                    // retrieve all its properties into an array, and iterate over them.
                    Type type = exif.GetType();
                    PropertyInfo[] properties = type.GetProperties();
                    foreach (PropertyInfo property in properties)
                    {
                        // Display property name and its value.
                        Console.WriteLine(property.Name + ":" + property.GetValue(exif, null));
                    }
                }
            }

            Console.WriteLine("Finished example ReadAndModifyJpegEXIFTags");
            //ExEnd:ReadAndModifyJpegEXIFTags
        }
    }
}